Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli

```diff
@@ -188,41 +188,90 @@ struct ClipmapLevel
+    // ignore the last level, as it will be used as a fall back default.
+    uint clipmapStackSize = TerrainSrg::m_clipmapData.m_macroClipmapStackSize - 1;
+    for (uint clipmapIndex = 0; clipmapIndex < clipmapStackSize; ++clipmapIndex)
+    {
+        float2 clipmapCenterInWorldSpace = GetMacroClipmapCenterInWorldSpace(clipmapIndex);
+        float clipmapToWorldScale = GetMacroClipmapToWorldScale(clipmapIndex);
+        float extent = (clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize) * clipmapToWorldScale;
 
-    float2 maxRenderSize = float2(maxRenderDistance, maxRenderDistance);
-    // The top clipmap's render distance
-    float2 minRenderSize = maxRenderSize / pow(clipmapScaleBase, (float)(clipmapStackSize - 1u));
+        float contains = AabbContainsPoint(clipmapCenterInWorldSpace, extent, worldPosition);
+        closestLevel += contains;
+    }
+    clipmapLevel.m_closestLevel = TerrainSrg::m_clipmapData.m_macroClipmapStackSize - uint(closestLevel);
+    clipmapLevel.m_nextLevel = clipmapLevel.m_closestLevel + 1;
 
-    float2 absDistance = abs(distanceFromViewPosition);
-    float2 clampedDistance = clamp(absDistance * RenderDistanceBias, minRenderSize, maxRenderSize);
-    float2 distanceRatio = maxRenderSize / clampedDistance;
-    // Due to clamping, the max result of log is (clipmapStackSize - 1u), which falls into the first clipmap
-    clipmapLevel.m_closestLevel = clipmapStackSize - 1u - uint(floor(log(min(distanceRatio.x, distanceRatio.y))/log(clipmapScaleBase)));
+    // Calculate the blending fact if the position falls into the blending band.
+    float2 clipmapCenterInWorldSpace = GetMacroClipmapCenterInWorldSpace(clipmapLevel.m_closestLevel);
+    float clipmapToWorldScale = GetMacroClipmapToWorldScale(clipmapLevel.m_closestLevel);
 
-    float renderDistance = maxRenderDistance / pow(clipmapScaleBase, (float)(clipmapStackSize - 1u - clipmapLevel.m_closestLevel));
-    float blendingStart = renderDistance * BlendingStartFactor;
-    float blendingEnd = renderDistance / RenderDistanceBias;
-    float blendingValue = max(absDistance.x, absDistance.y);
+    float blendBeginDistance = clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize - BetweenLevelsBlendBandSize;
+    // Clipmap uses Chebyshev distance
+    float2 distance = abs(worldPosition - clipmapCenterInWorldSpace);
+    float cDistance = max(distance.x, distance.y) / clipmapToWorldScale;
+    clipmapLevel.m_weight = max(0.0, (cDistance - blendBeginDistance) / BetweenLevelsBlendBandSize);
 
-    if (blendingValue > blendingStart && blendingValue < blendingEnd)
-    {
-        clipmapLevel.m_weight = (blendingValue - blendingStart) / (blendingEnd - blendingStart);
-        clipmapLevel.m_nextLevel = clipmapLevel.m_closestLevel + 1;
-    }
-    else
+    return clipmapLevel;
+}
+
+ClipmapLevel CalculateDetailClipmapLevel(float2 worldPosition)
+{
+    uint clipmapSize = TerrainSrg::m_clipmapData.m_clipmapSizeUint;
```

Comment:
You're using the uint version here even though you quickly cast it to a float with / 2.0.